crates/postgres/src/store.rs

```diff
@@ -0,0 +1,926 @@
+use async_trait::async_trait;
+use bdk_chain::bitcoin::consensus::{deserialize, serialize};
+use bdk_chain::bitcoin::hashes::Hash;
+use bdk_chain::bitcoin::{Amount, Network, OutPoint, ScriptBuf, Transaction, TxOut};
+use bdk_chain::bitcoin::{BlockHash, Txid};
+use bdk_chain::miniscript::descriptor::{Descriptor, DescriptorPublicKey};
+use serde::{Deserialize, Serialize};
+use sqlx::types::JsonValue;
+use sqlx::{Pool, Postgres, Row};
+use std::collections::{BTreeMap, BTreeSet};
+use std::fmt::Debug;
+use std::marker::PhantomData;
+use std::str::FromStr;
+use std::sync::Arc;
+
+use crate::Error;
+use bdk_chain::persist::CombinedChangeSet;
+use bdk_chain::{
+    indexed_tx_graph, keychain, local_chain, tx_graph, Anchor, Append, DescriptorExt, DescriptorId,
+};
+
+/// Persists data in to a relational schema based [Postgres] database file.
+///
+/// The changesets loaded or stored represent changes to keychain and blockchain data.
+///
+/// [Postgres]: https://www.postgresql.org
+pub struct Store<K, A> {
+    // A sqlx connection to the Postgres database.
```

Comment:
```suggestion
    /// A sqlx connection to the Postgres database.
```

This could be a docstring